packages/eslint-plugin/src/rules/use-default-type-parameter.ts

```diff
@@ -0,0 +1,161 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import * as tsutils from 'tsutils';
+import ts from 'typescript';
+import * as util from '../util';
+import { findFirstResult } from '../util';
+
+interface ArgsAndParams {
+  typeArguments: ts.NodeArray<ts.TypeNode>;
+  typeParameters: readonly ts.TypeParameterDeclaration[];
+}
+
+type ExtendingClassLikeDeclaration = ts.ClassLikeDeclaration & {
+  heritageClauses: ts.NodeArray<ts.HeritageClause>;
+};
+
+type ParameterCapableTSNode =
+  | ts.CallExpression
+  | ts.NewExpression
+  | ts.TypeReferenceNode
+  | ts.ExpressionWithTypeArguments;
+
+type MessageIds = 'unnecessaryTypeParameter';
+
+export default util.createRule<[], MessageIds>({
+  name: 'use-default-type-parameter',
+  meta: {
+    docs: {
+      description:
+        'Warns if an explicitly specified type argument is the default for that type parameter',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    fixable: 'code',
+    messages: {
+      unnecessaryTypeParameter:
+        'This is the default value for this type parameter, so it can be omitted.',
+    },
+    schema: [],
+    type: 'suggestion',
+  },
+  defaultOptions: [],
+  create(context) {
+    const parserServices = util.getParserServices(context);
+    const checker = parserServices.program.getTypeChecker();
+
+    function checkTSArgsAndParameters(
+      esParameters: TSESTree.TSTypeParameterInstantiation,
+      { typeArguments, typeParameters }: ArgsAndParams,
+    ): void {
+      // Just check the last one. Must specify previous type parameters if the last one is specified.
+      const i = typeArguments.length - 1;
+      const arg = typeArguments[i];
+      const param = typeParameters[i];
+
+      // TODO: would like checker.areTypesEquivalent. https://github.com/Microsoft/TypeScript/issues/13502
```

Comment:
could we just introduce https://github.com/runem/ts-simple-type#readme ?

Reply:
In theory, sure. Would you like me to do this here or in a separate PR?